### PR TITLE
separate storage cluster in it's own services

### PR DESCRIPTION
**Pull request by khaledkbadr**
## Don't Merge, still in testing phase
related to #1137

### CODE REVIEW

File: api/healthcheck/health_api_StorageCluster.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func (api *HealthCheckApi) ListStorageClusterHealth(w http.ResponseWriter, r *ht
 	vars := mux.Vars(r)
 	storageClusterID := vars["storageclusterid"]
 
-	serviceName := fmt.Sprintf("storage_cluster_%s", storageClusterID)
+	serviceName := fmt.Sprintf("storagecluster.block_%s", storageClusterID)
```
Comment (serboctor) [inline api/healthcheck/health_api_StorageCluster.go]: Remove the role.

File: api/healthcheck/health_api_StorageClusters.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func (api *HealthCheckApi) ListStorageClustersHealth(w http.ResponseWriter, r *h
 		}
 
 		// Get healthcheck data
-		serviceName := fmt.Sprintf("storage_cluster_%s", service.Name)
+		serviceName := fmt.Sprintf("storagecluster.block_%s", service.Name)
```
Comment (serboctor) [inline api/healthcheck/health_api_StorageClusters.go]: Remove role

File: pyclient/zeroos/orchestrator/sal/StorageCluster.py [hunk 3]
```diff
@@ -2,76 +2,33 @@
...
 
 
-class StorageCluster:
-    """StorageCluster is a cluster of StorageEngine servers"""
-
-    def __init__(self, label, nodes=None, disk_type=None, logger=None):
-        """
-        @param label: string repsenting the name of the storage cluster
-        """
+class StorageClusterBasic:
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/StorageCluster.py]: StorageClusterBasic sounds a bit weird. I prefer something like BaseStorageCluster, it sounds more standard.

File: templates/vdisk_image/actions.py [hunk 4]
```diff
@@ -48,7 +45,7 @@ def install(job):
             cmd += ' --force'
 
         if vdiskstore.model.data.objectCluster:
-            storageclusterservice = service.aysrepo.serviceGet(role='storage_cluster',
+            storageclusterservice = service.aysrepo.serviceGet(role='storagecluster.object',
```
Comment (serboctor) [inline templates/vdisk_image/actions.py]: shouldn't this be `actor=` instead of `role=`

Comment (serboctor) [reply templates/vdisk_image/actions.py]: I know that role is only `storagecluster` but I am actually not sure if role allows filtering using the actor.

File: templates/vdisk_image/actions.py [hunk 5]
```diff
@@ -152,22 +149,15 @@ def save_config(job):
...
 
-    cluster = vdiskstore.model.data.blockCluster if type == "block" else vdiskstore.model.data.objectCluster
+    cluster = vdiskstore.model.data.blockCluster
 
-    storageclusterservice = service.aysrepo.serviceGet(role='storage_cluster',
-                                                       instance=cluster)
-    cluster = StorageCluster.from_ays(storageclusterservice, job.context['token'])
-    nodes = list(set(storageclusterservice.producers["node"]))
-    return {"config": cluster.get_config(), "nodes": nodes, 'dataShards': cluster.data_shards, 'parityShards': cluster.parity_shards}
+    blockcluster_service = service.aysrepo.serviceGet(role='storagecluster.block', instance=cluster)
```
Comment (serboctor) [inline templates/vdisk_image/actions.py]: Same as the previous comment.

File: templates/vdiskstorage/actions.py [hunk 6]
```diff
@@ -1,20 +1,11 @@
-from js9 import j
-
 def input(job):
     service = job.service
     args = job.model.args
-    block_cluster_service = service.aysrepo.serviceGet(instance=args.get('blockCluster'), role='storage_cluster')
-    if 'block' != block_cluster_service.model.data.clusterType:
-        raise RuntimeError('storage_cluster %s provided should be of type  block' % block_cluster_service.name)
+    block_service = service.aysrepo.serviceGet(instance=args.get('blockCluster'), role='storagecluster.block')
```
Comment (serboctor) [inline templates/vdiskstorage/actions.py]: Same as the previous comment.

File: templates/vdiskstorage/actions.py [hunk 7]
```diff
@@ -1,20 +1,11 @@
...
     if args.get('objectCluster'):
-        object_cluster_service = service.aysrepo.serviceGet(instance=args.get('objectCluster'), role='storage_cluster')
-        if 'object' != object_cluster_service.model.data.clusterType:
-            raise RuntimeError('storage_cluster %s provided should be of type  object' % object_cluster_service.name)
         if args.get('slaveCluster'):
-            slave_cluster_service = service.aysrepo.serviceGet(instance=args.get('slaveCluster'), role='storage_cluster')
-            if 'block' != slave_cluster_service.model.data.clusterType:
-                raise RuntimeError('storage_cluster %s provided should be of type  block' % slave_cluster_service.name)
-            if block_cluster_service.model.data.nrServer < slave_cluster_service.model.data.nrServer:
+            slave_cluster_service = service.aysrepo.serviceGet(instance=args.get('slaveCluster'), role='storagecluster.block')
```
Comment (serboctor) [inline templates/vdiskstorage/actions.py]: Same

File: api/healthcheck/health_api_StorageCluster.go (hunk 1 above)
Comment (khaledkbadr) [reply api/healthcheck/health_api_StorageCluster.go]: actually this is the service name, but you're right a `.` is confusing here

File: api/healthcheck/health_api_StorageClusters.go (hunk 2 above)
Comment (khaledkbadr) [reply api/healthcheck/health_api_StorageClusters.go]: same

File: templates/vdisk_image/actions.py (hunk 4 above)
Comment (khaledkbadr) [reply templates/vdisk_image/actions.py]: There's no actor in serviceGet, role works both ways here

File: templates/vdisk_image/actions.py (hunk 5 above)
Comment (khaledkbadr) [reply templates/vdisk_image/actions.py]: same :D

File: templates/vdiskstorage/actions.py (hunk 7 above)
Comment (khaledkbadr) [reply templates/vdiskstorage/actions.py]: same

File: templates/vdiskstorage/actions.py (hunk 6 above)
Comment (khaledkbadr) [reply templates/vdiskstorage/actions.py]: same :D

**Pull request merged**